setup_unit_tests.py

```diff
@@ -0,0 +1,488 @@
+                                data_dir: Optional[Path], metadata: Union[List[Any], Dict[str, Any]]):
+    """
+    Get the expected output values with respect to the output types.
+    This is done to get the md5sum of files when converting unit tests.
+
+    Ex: If given hello.txt and type File, try to get the regex of the file
+    If given "hello" and type String, return string
+    """
+    if output_values is None and output_type.optional:
+        return output_values
+    if isinstance(output_type, WDL.Type.File):
+        # this will only populate md5sum if the files exist recursively in the data directory
+        for path in glob.glob(str(data_dir / "**"), recursive=True):
+            if path.endswith(output_values) and os.path.exists(path):
+                with open(path, "rb") as f:
+                    md5sum = hashlib.md5(f.read()).hexdigest()
+                return {'md5sum': md5sum}
+            # else check if there is extra metadata to read from
+            if get_not_none(metadata, "md5sum") is not None:
+                return {'md5sum': metadata.get("md5sum")}
+            if get_not_none(metadata, "regex") is not None:
+                return {'regex': metadata.get("regex")}
+
+    if isinstance(output_type, WDL.Type.StructInstance):
+        converted_output = dict()
+        for i, (output_key, output_value) in enumerate(output_values.items()):
+            try:
+                output_value_type = output_type.members[output_key]
+            except KeyError:
+                # coercion from map to struct can cause some weird type behavior
+                # since dictionaries past python 3.6 are ordered, find the corresponding type from the current output's position
+                output_value_type = list(output_type.members.values())[i]
+            converted_output[output_key] = convert_typed_output_values(output_value, output_value_type, data_dir, get_not_none(metadata, output_value))
+        return converted_output
+    if isinstance(output_type, WDL.Type.Map):
+        converted_output = dict()
+        for output_key, output_value in output_values.items():
+            new_output_key = convert_typed_output_values(output_key, output_type.item_type[0], data_dir, get_not_none(metadata, output_key))
+            new_output_value = convert_typed_output_values(output_value, output_type.item_type[1], data_dir, get_not_none(metadata, output_value))
+            converted_output[new_output_key] = new_output_value
+    if isinstance(output_type, WDL.Type.Pair):
+        converted_output = dict()
+        # key should be left or right
+        converted_output["left"] = convert_typed_output_values(output_values["left"], output_type.left_type, data_dir, get_not_none(metadata, output_values["left"]))
+        converted_output["right"] - convert_typed_output_values(output_values["right"], output_type.right_type, data_dir, get_not_none(metadata, output_values["right"]))
```

Comment:
I think the `-` is probably meant to be an `=`?

Reply:
Fixed